test/test_local_io.py

```diff
@@ -655,6 +655,42 @@ def filepath_fn(name: str) -> str:
         # __len__ Test: returns the length of source DataPipe
         self.assertEqual(3, len(saver_dp))
 
+    @skipIfNoIoPath
+    def test_io_path_saver_file_lock(self):
+        def filepath_fn(name: str) -> str:
+            return os.path.join(self.temp_dir.name, os.path.basename(name))
+
+        # Same filename with different name
+        name_to_data = {"1.txt": b"DATA1", "1.txt": b"DATA2", "2.txt": b"DATA3", "2.txt": b"DATA4"}  # noqa: F601
+
+        # Add sharding_filter to shard data into 2
+        source_dp = IterableWrapper(name_to_data.items()).sharding_filter()
```

Comment:
Based on the failure on workflow, could you please fix:
```suggestion
        source_dp = IterableWrapper(list(name_to_data.items())).sharding_filter()
```